Update Vim Config to use .vimrc

DominoTree opened the pull request:
These changes allow `.vimrc` to be used for settings instead of `~/.vim/after/ftplugin/rust.vim` - I think most people keep all of their vim settings in one file.

Also, this quiets down rusty-tags more because it was still printing output into my Vim session.

dan-t commented:
> I think most people keep all of their vim settings in one file.

I don't quite buy this argument, because if someone uses
a plugin, then there's already a `~/.vim` directory.

I'm not a fan of putting everything into the `~/.vimrc` so
I don't want to propose it alone. 

I'm fine with having it as an additional configuration info,
but in this case it's more vim idiomatic to have `autocmd FileType rust ...`.

> Also, this quiets down rusty-tags more because it was still printing output into my Vim session.

There's the `--quiet` option which should be used in the `README`.

DominoTree commented:
This is just information that was useful to me - I didn't expect the default configuration example to write output across my terminal, and I've been using Vim for years and have never seen `~/.vim/after/` or `~/.vim/after/ftplugin/` used anywhere that I can recall, so I just thought this might be useful for other people too :)

dan-t commented:
Ok, I think you're right. People knowing about the vim filetype system
most likely wont have issues with the configuration. So making
the README more accessable for newbies is a good idea.

Can you please change the redirection to `/dev/ null` by using
the option `--quiet`?

Then I will merge the changes. Thanks!

DominoTree commented:
I switched the redirection and switched the autocmds to use FileType as you suggested :)

Could probably add a statement to check whether $RUST_SRC_PATH is set and if so append it to the tags variable, but I'm not sure how helpful that would be

dan-t commented:
Mixing `FileType rust` and `*.rs` doesn't feel right, and combing `FileType rust` with `BufWrite` is cumbersome, so I would just go with `*.rs`, which also reduces a dependency on an available
rust language plugin.

Can you please make this change? Thanks!

dan-t commented:
Thanks!
